File: 4_median_of_two_arrays/median_of_two_arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        arr3 = []
        m, n = len(nums1), len(nums2)
        i = j = 0

        while i < m and j < n:
            if nums1[i] < nums2[j]:
                arr3.append(nums1[i])
                i += 1
            else:
                arr3.append(nums2[j])
                j += 1

        arr3.extend(nums1[i:])
        arr3.extend(nums2[j:])

        t = m + n
        if t % 2 == 1:
            return arr3[t // 2]
        
        return (arr3[t // 2] + arr3[t // 2 - 1]) / 2.0

File: 4_median_of_two_arrays/median_of_two_arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> arr;
        for(int i = 0; i<nums1.size();i++){
            arr.push_back(nums1[i]);
        }
        for(int j=0;j<nums2.size();j++){
            arr.push_back(nums2[j]);
        }
        sort(begin(arr),end(arr));
        int t = arr.size();
        if(t % 2 == 1){
            return arr[t / 2];
        }
        return (arr[t / 2] + arr[t / 2 - 1]) / 2.0;
    }
};